src/orquestra/sdk/_base/_api/_wf_run.py

```diff
@@ -444,6 +508,17 @@ def get_tasks(self) -> t.Set[TaskRun]:
                 wf_def=self._wf_def,
             )
             for task_run_model in wf_run_model.task_runs
+            if matches_model_filters(
+                task_run_model,
+                state=state,
+                task_run_id=task_run_id,
+                task_invocation_id=task_invocation_id,
+            )
+        }
+        return {
+            task
+            for task in tasks
+            if matches_run_filters(task, task_fn_name=function_name)
```

Comment:
I see we're filtering on schema TaskRun and api TaskRun.

It's a bit confusing on first read, is there a way we can filter in one go (even with multiple filter functions)

Reply:
New version: 
```python
return {
    task
    for task_model in wf_run_model.task_runs
    if self._task_matches_schema_filters(
        task_model,
        state=state,
        task_run_id=task_run_id,
        task_invocation_id=task_invocation_id,
    )
    and self._task_matches_api_filters(
        task := TaskRun(
            task_run_id=task_model.id,
            task_invocation_id=task_model.invocation_id,
            workflow_run_id=self.run_id,
            runtime=self._runtime,
            wf_def=self._wf_def,
        ),
        task_fn_name=function_name,
    )
}
```

WDYT?

Reply:
I take it back, this makes it even harder to read!

Reply:
We should definitely aim for readable, so what's your preference out of:
1. Leave it as-is and add some comments that explain what's going on
2. Revert to the old version and add comments
3. Turn the listcomp into a for loop, something like:
```python
tasks = set()
for task_model in wf_run_model.task_runs:
    if self._task_matches_schema_filters(task_model, ...):
       task = TaskRun(...)
       if self._task_matches_api_filters(task, ...):
          tasks.add(task)
return tasks
```
?

Reply:
Discussed on slack!